src/theory/quantifiers/trigger.h

```diff
@@ -35,10 +35,19 @@ class InstMatchGenerator;
 
 /** Information about a node used in a trigger. 
 *
+* This information includes:
+* 1. the free variables of the node, and
+* 2. information about which terms are useful for matching.
+*
 * As an example, consider the trigger 
 *   { f( x ), g( y ), P( y, z ) } 
 * for quantified formula 
 *   forall xy. ( f( x ) != b => ( P( x, g( y ) ) V P( y, z ) ) )
+*
+* Notice that it is only useful to match f( x ) to terms not in the equivalence class of b,
+* and P( y, z ) to terms in the equivalence class of false, as such instances will always be 
```

Comment:
By "instance", I mean the formula that is the result of applying a substitution to the body of the quantified formula.

So any result of E-matching that matched f( x ) with a term f( s ) in the equivalence class of b is such that the instance : 
( f( s ) != b => ( P( t, g( y ) ) V P( y, z ) ) )
for any extension of that substitution for y,z is entailed by the current set of ground equalities and disequalities E since 
E |= f( s ) = b, and hence
E |= ( f( s ) != b => ( P( t, g( y ) ) V P( y, z ) ) ).
CVC4 does not add any instance that is currently entailed by E.